## Assignment 3
### A good Null/alternative hypothesis statement and treatment, with a clear Null and Alternative spelled out AND written out as a formula, and a good interpretation of the Z value you obtain in terms of ability or inability to reject the Null Hypothesis. 

## Question: How is the new bus route for X8 improving commute times?
### Null Hypothesis: The mean commute time for the new bus route (Tnew) is the same or longer than the old bus route (Told).
### Alternate Hypothesis: The mean commute time for the new bus route (Tnew) is shorter than the old bus route (Told).
### Formula
### H0: Tnew - Told >= 0    
### Ha: Tnew - Told < 0
  ###  α = 0.05

In [16]:
from __future__ import print_function
__author__ = 'Chunqing Xu CUSP 2016'
import csv
import requests
import sys
import os

import numpy as np
from numpy.random import randn
import pylab as pl
import pandas as pd

In [17]:
#Given old bus route data: ~N(u=36,o=6)
Told_mean = 36 #mean travel time of old bus route
Told_std = 6

#busdata = new bus route travel times
data = !curl -O https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt
busdata = pd.read_csv('times.txt', sep=" ", header = None, )

#New bus route mean
Tnew_mean = busdata.mean()
#busdata.describe()
Tnew_mean

0    34.466162
dtype: float64

### Tnew or the new average commute time is 34.466162 which is less than the average of the old bus route.

In [18]:
#new bus route standard deviation
Tnew_std = busdata.std()

### If p(Ha)-p(Ho) > α the null hypothesis H0 is falsified at the 1-α confidence level
### Z Score Formula: z = (x – μ) / (σ / √n) Z ~ (mean(pop) - mean(sample)) / (stdev / sqrt(N))

In [19]:
z_score = (Told_mean-Tnew_mean)/(Told_std/np.sqrt(len(busdata)))
z_score

0    2.556397
dtype: float64

### Since the z score is more than 2 standard deviation from the mean, surpassing the threshold, we can reject the null hypothesis. The outcome indicates that average commute time for the new bus route is shorter; statistically significant at the 95% confidence level.